woocommerce/woocommerce-ios
# Shipping Labels: update the parsing logic of Carriers and Rates for managing multi-packages

Part of #4716

## Description
Since we are going to support the multi-packages in Shipping Labels, we need to update the logic for parsing Carriers and Rates from the endpoint `/rates`.
In this PR, I updated the `ShippingLabelCarriersAndRatesMapper`, which now will parse every single key which will have the `id` of the package (generated using UUID). The mapper now will return an array of carriers and rates for every single package.
Every `ShippingLabelCarriersAndRates` now will contain the package id, which will be used inside the Carrier and Rates screen in a later PR.
I also updated the logic in `ShippingLabelCarriersViewModel` for generating the rows of the view, because now we need to iterate around the array of carriers.

The new keys that we are going to parse to generate the `ShippingLabelCarriersAndRates` array are highlighted in the image below. Each of those keys represents the rates of each individual package passed to the endpoint.
<img width="1030" alt="Schermata 2021-09-30 alle 12 17 37" src="https://user-images.githubusercontent.com/495617/135437652-34129f54-f3ea-43d7-a75a-079529943496.png">

## Testing
1. Turn off the feature flag `shippingLabelsMultiPackage` under `DefaultFeatureFlagService`.
2. Try to create a new shipping label.
3. The Carriers and Rates screen should work like before.

Update release notes:
